docker/api/container.py

```diff
@@ -940,8 +941,25 @@ def put_archive(self, container, path, data):
         """
         params = {'path': path}
         url = self._url('/containers/{0}/archive', container)
-        res = self._put(url, params=params, data=data)
-        self._raise_for_status(res)
+        try:
+            res = self._put(url, params=params, data=data)
+        except requests.exceptions.ConnectionError as e:
```

Comment:
The problem here is that you're making the assumption that the only reason a `ConnectionError` could occur is because of an incorrect path issue. This will result in a lot of "false positives", misidentifying the issue and misleading the user.

The better fix would be to figure out why the ConnectionError is raised in the first place, and fix the cause.

Reply:
Agree. We need more fine-grained control here. I have some ideas how to track the exact behavior described in #1808. Will experiment a bit and come up with improvements.